gitlab/resource_gitlab_project_approval_rule_test.go

```diff
@@ -0,0 +1,306 @@
+
+		if !reflect.DeepEqual(userNames, want.ApproverUsernames) {
+			return fmt.Errorf("got approvers %s; want %s", userNames, want.ApproverUsernames)
+		}
+
+		// Compare unique group paths
+		var groupPaths []string
+		for _, group := range projectApprovalRule.Groups {
+			groupPaths = append(groupPaths, group.Path)
+		}
+		sort.Strings(groupPaths)
+
+		if !reflect.DeepEqual(groupPaths, want.GroupPaths) {
+			return fmt.Errorf("got groups %s; want %s", groupPaths, want.GroupPaths)
+		}
+
+		return nil
+	}
+}
+
+func testAccGitLabProjectApprovalRuleCreateConfig(
+	randomInt int,
+	approvals int,
+	userIDs string,
+	groupIDs string,
+) string {
+	return fmt.Sprintf(`
+resource "gitlab_project_approval_rule" "foo" {
+	project            = gitlab_project.foo.id
+	name               = "foo rule %d"
+	approvals_required = %d
+	user_ids           = [%s]
+	group_ids          = [%s]
+}
+
+resource "gitlab_project" "foo" {
+	name              = "foo project"
+	path              = "foo-project-%d"
+	description       = "Terraform acceptance test - Approval Rule"
+	visibility_level  = "public"
+}
+
+resource "gitlab_project_membership" "baz" {
+  project_id     = gitlab_project.foo.id
+  user_id        = gitlab_user.baz.id
+  access_level   = "developer"
+}
+
+resource "gitlab_project_membership" "qux" {
+  project_id     = gitlab_project.foo.id
+  user_id        = gitlab_user.qux.id
+  access_level   = "developer"
+}
+
+resource "gitlab_group" "foo" {
+	name             = "foo-group"
+	path             = "foo-group-%d"
+	description      = "Terraform acceptance tests - Approval Rule"
+	visibility_level = "public"
+}
+
+resource "gitlab_group" "bar" {
+	name             = "bar-group"
+	path             = "bar-group-%d"
+	description      = "Terraform acceptance tests - Approval Rule"
+	visibility_level = "public"
+}
+
+resource "gitlab_user" "foo" {
+	name             = "foo user"
+	username         = "foo-user-%d"
+	password         = "foo12345"
+	email            = "foo-user@ssss.com"
+}
+
+resource "gitlab_user" "bar" {
+	name             = "bar user"
+	username         = "bar-user-%d"
+	password         = "bar12345"
+	email            = "bar-user@ssss.com"
+}
+
+resource "gitlab_user" "baz" {
+	name             = "baz user"
+	username         = "baz-user-%d"
+	password         = "baz12345"
+	email            = "baz-user@ssss.com"
+}
+
+resource "gitlab_user" "qux" {
+	name             = "qux user"
+	username         = "qux-user-%d"
+	password         = "qux12345"
+	email            = "qux-user@ssss.com"
+}
+
+resource "gitlab_group_membership" "foo" {
+  group_id         = gitlab_group.foo.id
+  user_id          = gitlab_user.foo.id
+  access_level     = "developer"
+}
+
+resource "gitlab_group_membership" "bar" {
+  group_id        = gitlab_group.bar.id
+  user_id         = gitlab_user.bar.id
+  access_level    = "developer"
+}
+	`,
+		randomInt, // "foo rule %d"
+		approvals,
+		userIDs,
+		groupIDs,
+		randomInt, // foo-project-%d
```

Comment:
For future reference (don't worry about changing this now), see: https://stackoverflow.com/questions/26624699/is-there-a-way-to-reuse-an-argument-in-fmt-printf